src/core/util/byte_stream.cc

```diff
@@ -100,6 +101,13 @@ void OutputByteStream::ProduceData(std::size_t amount) {
 }
 
 void OutputByteStream::WriteData(const std::uint8_t* data, std::size_t len) {
+  if (!len)
```

Comment:
I still strongly disagree with not throwing here (I'm not yet convinced as to why we should allow 0 length data) but, since you've provided debug logging, I'll pick and choose my battles 😆